coalib/bearlib/abstractions/Linter.py

```diff
@@ -53,6 +53,17 @@ def _prepare_options(options, bear_class):
             assert_right_type(options['diff_distance'], int, 'diff_distance')
 
         allowed_options |= {'diff_severity', 'result_message', 'diff_distance'}
+    elif options['output_format'] == 'unidiff':
+        if (
+                'diff_severity' in options and
+                options['diff_severity'] not in RESULT_SEVERITY.reverse):
+            raise TypeError('Invalid value for `diff_severity`: ' +
+                            repr(options['diff_severity']))
+
+        if 'result_message' in options:
+            assert_right_type(options['result_message'], str, 'result_message')
+
+        allowed_options |= {'diff_severity', 'result_message'}
```

Comment:
the code is mostly duplicated with the if-stmt before, could you try to merge those?